### DISCUSSION TITLE
UML diagrams for cuda-quantum impl design

### DISCUSSION THREAD

**Discussion by stevescia** [NVIDIA/cuda-quantum#93] [Ideas]
Wanted feedback on documenting the cuda-quantum obect model with UML drawings; may be useful to people learning the impl details of cuda-quantum. I often draw uml diagrams as a way to learn an object model (the exercise of actual drawing the diagrams helps sink the details in, as well as serving as a reference point later); Started doing this for xacc, but now switched to cuda-quantum.

ofc generating these from code would be ideal.

here's a screen shot of an initial plant uml class diagram for qudit and qreg. comments on the diagrams, content, style welcome
![screenshot_qudit_qreg](https://user-images.githubusercontent.com/11952339/232898202-51ef96f4-6755-4ba3-b34c-b56d511977b6.png)

**Comment by sachingodishela** [comment]
Nice. We can have these diagrams in README files in relevant folders. Checkout [mermaid](https://mermaid.js.org/syntax/classDiagram.html), which is supported by GitHub in markdown files. See examples [here](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-diagrams).
